# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/sid.h
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/sid.h
--- a/security/sandbox/chromium/sandbox/win/src/sid.h	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/sid.h	Wed Jun 12 11:10:48 2019 +0100
@@ -7,20 +7,63 @@
 
 #include <windows.h>
 
+#include "base/strings/string16.h"
+
 namespace sandbox {
 
+// Known capabilities defined in Windows 8.
+enum WellKnownCapabilities {
+  kInternetClient,
+  kInternetClientServer,
+  kPrivateNetworkClientServer,
+  kPicturesLibrary,
+  kVideosLibrary,
+  kMusicLibrary,
+  kDocumentsLibrary,
+  kEnterpriseAuthentication,
+  kSharedUserCertificates,
+  kRemovableStorage,
+  kAppointments,
+  kContacts,
+  kMaxWellKnownCapability
+};
+
 // This class is used to hold and generate SIDS.
 class Sid {
  public:
+  // As PSID is just a void* make it explicit.
+  explicit Sid(PSID sid);
   // Constructors initializing the object with the SID passed.
   // This is a converting constructor. It is not explicit.
-  Sid(const SID *sid);
+  Sid(const SID* sid);
   Sid(WELL_KNOWN_SID_TYPE type);
 
+  // Create a Sid from an AppContainer capability name. The name can be
+  // completely arbitrary.
+  static Sid FromNamedCapability(const wchar_t* capability_name);
+  // Create a Sid from a known capability enumeration value. The Sids
+  // match with the list defined in Windows 8.
+  static Sid FromKnownCapability(WellKnownCapabilities capability);
+  // Create a Sid from a SDDL format string, such as S-1-1-0.
+  static Sid FromSddlString(const wchar_t* sddl_sid);
+  // Create a Sid from a set of sub authorities.
+  static Sid FromSubAuthorities(PSID_IDENTIFIER_AUTHORITY identifier_authority,
+                                BYTE sub_authority_count,
+                                PDWORD sub_authorities);
+  // Create the restricted all application packages sid.
+  static Sid AllRestrictedApplicationPackages();
+
   // Returns sid_.
-  const SID *GetPSID() const;
+  PSID GetPSID() const;
+
+  // Gets whether the sid is valid.
+  bool IsValid() const;
+
+  // Converts the SID to a SDDL format string.
+  bool ToSddlString(base::string16* sddl_string) const;
 
  private:
+  Sid();
   BYTE sid_[SECURITY_MAX_SID_SIZE];
 };
 